src/stress.jl

```diff
@@ -50,35 +50,35 @@ Reference:
 """
 module Stress
 
-using GeometryTypes, Compat, FixedSizeArrays
+using GeometryTypes, Compat, StaticArrays
 import Base: start, next, done, *
 
-function (*){T<:LinAlg.BlasFloat,S<:FixedArray}(A::StridedMatrix{T}, x::StridedVector{S})
+function (*){T<:LinAlg.BlasFloat,S<:StaticArray}(A::StridedMatrix{T}, x::StridedVector{S})
     A_mul_B!(similar(x, S, size(A,1)), A, x)
 end
 
-immutable Layout{M1<:AbstractMatrix, M2<:AbstractMatrix, VP<:AbstractVector,FT<:AbstractFloat}
+immutable Layout{M1<:AbstractMatrix, M2<:AbstractMatrix, VP<:AbstractVector, FT<:AbstractFloat}
     δ::M1
     weights::M2
     positions::VP
-    pinvLw::Matrix{FT}
+    pinvLw::Matrix{Float64}
     iterations::Int
     abstols::FT
     reltols::FT
     abstolx::FT
 end
 
 
-function initialweights(D, T=eltype(D))
+function initialweights(D)
     map(D) do d
-        x = T(d^(-2.0))
-        isfinite(x) ? x : zero(T)
+        x = Float64(d^(-2.0))
```

Comment:
Sorry, just jumping in here - haven't read the entire comment thread but wanted to make sure you knew that `adjacency_matrix(::AbstractGraph, ::DataType)` was an option:
```
julia> g = Graph(3,3)
{3, 3} undirected simple Int64 graph

julia> adjacency_matrix(g, Float64)
3×3 SparseMatrixCSC{Float64,Int64} with 6 stored entries:
  [2, 1]  =  1.0
  [3, 1]  =  1.0
  [1, 2]  =  1.0
  [3, 2]  =  1.0
  [1, 3]  =  1.0
  [2, 3]  =  1.0
```

again, not sure if relevant or helpful, but there it is.